### PR TITLE
use ansible style patterns for commcare cloud commands

### PR DESCRIPTION
**Pull request by snopoke**
It always confuses me that we have different pattern styles for commcare-cloud and ansible:
```
cchq <env> ssh webworkers:2

# vs

cchq <env> aps --limit webworkers[2]
```

This deprecates the `:x` style host selector for commcare-cloud in favour of `[x]`. The old style will still work with a warning:
```
$ cchq icds ssh ansible@webworkers:0
[WARNING]: Use '[x]' to select hosts instead of ':x' which has been deprecated.
....
```

### CODE REVIEW

File: src/commcare_cloud/commands/inventory_lookup/inventory_lookup.py [hunk 1]
```diff
@@ -73,7 +76,8 @@ class Ssh(_Ssh):
     """
 
     def run(self, args, ssh_args):
-        if args.server.split(':')[0] == 'control' and '-A' not in ssh_args:
+
+        if 'control' in split_host_group(args.server).group and '-A' not in ssh_args:
```
Comment (snopoke) [inline src/commcare_cloud/commands/inventory_lookup/inventory_lookup.py:80]: this now works even if you are specifying the user `user@control`

File: docs/commcare-cloud/commands/index.md [hunk 2]
```diff
@@ -161,7 +161,7 @@ multiple servers if there is more than one in the group. For
 example: webworkers:0 will pick the first webworker. May also be
```
Comment (millerdev) [inline docs/commcare-cloud/commands/index.md:161]: Should `webworkers:0` to be changed to `webworkers[0]`? (Other parts of this paragraph need to be rewritten too if yes.) Same below on line 189, 218, etc. There's another similar paragraph at https://github.com/dimagi/commcare-cloud/blob/6cb734133525f96cf8bdf0da3a680ab686c91a3a/src/commcare_cloud/commands/inventory_lookup/inventory_lookup.py#L27-L29

**Comment by dannyroberts** [comment]
Another little fix related to this: https://github.com/dimagi/commcare-cloud/pull/3985

File: docs/commcare-cloud/commands/index.md (hunk 2 above)
Comment (snopoke) [reply docs/commcare-cloud/commands/index.md:161]: https://github.com/dimagi/commcare-cloud/pull/3987

**Pull request merged**